azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -322,6 +355,44 @@ func validateAzureRMKustoClusterName(v interface{}, k string) (warnings []string
 	return warnings, errors
 }
 
+func expandOptimizedAutoScale(d *schema.ResourceData) *kusto.OptimizedAutoscale {
+	if v, ok := d.GetOk("optimized_auto_scale"); ok {
+		optimizedAutoScaleList := v.([]interface{})
+		optimizedAutoScaleMap := optimizedAutoScaleList[0].(map[string]interface{})
+		minimumInstances := int32(optimizedAutoScaleMap["minimum_instances"].(int))
+		maximumInstances := int32(optimizedAutoScaleMap["maximum_instances"].(int))
+
+		var version int32 = 1
+		var enabled bool = true
+
+		optimizedAutoScale := &kusto.OptimizedAutoscale{
+			Version:   &version,
+			IsEnabled: &enabled,
+			Minimum:   &minimumInstances,
+			Maximum:   &maximumInstances,
+		}
+
+		return optimizedAutoScale
+	}
+
+	return nil
+}
+
+func flattenOptimizedAutoScale(optimizedAutoScale *kusto.OptimizedAutoscale) []interface{} {
+	if optimizedAutoScale == nil {
+		return []interface{}{}
+	}
+
+	oas := map[string]interface{}{
+		"version":           int(*optimizedAutoScale.Version),
+		"enabled":           *optimizedAutoScale.IsEnabled,
+		"minimum_instances": int(*optimizedAutoScale.Minimum),
+		"maximum_instances": int(*optimizedAutoScale.Maximum),
```

Comment:
all 4 of these fields are nilable (and could be nil, so crash when accessing due to a bad API response) - as such can we make this:

```
enabled := false
if optimizedAutoScale.IsEnabled != nil {
	enabled = *optimizedAutoScale.IsEnabled
}

maxInstances := 0
if optimizedAutoScale.Maximum != nil {
	maxInstances = *optimizedAutoScale.Maximum
}

minInstances := 0
if optimizedAutoScale.Minimum != nil {
	minInstances = *optimizedAutoScale.Minimum
}

version := 0
if optimizedAutoScale.Version != nil {
	version = int(*optimizedAutoScale.Version)
}

return []interface{}{
	map[string]interface{}{
		"enabled": enabled,
		"maximum_instances": maxInstances,
		"minimum_instances": minInstances,
		"version": version,
	}
}
```